internal/pkg/service/cli/cmd/remote/table/download/cmd.go

```diff
@@ -125,3 +134,16 @@ func Command(p dependencies.Provider) *cobra.Command {
 
 	return cmd
 }
+
+func getColumns(ctx context.Context, d dependencies.RemoteCommandScope, f Flags, key keboola.TableKey) ([]string, error) {
+	if f.Header.IsSet() && len(f.Columns.Value) == 0 {
+		table, err := d.KeboolaProjectAPI().GetTableRequest(key).Send(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		return table.Columns, nil
+	}
+
+	return f.Columns.Value, nil
```

Comment:
```suggestion
	if !f.Header.IsSet() || len(f.Columns.Value) != 0 {
		return f.Columns.Value, nil
	}
	
	table, err := d.KeboolaProjectAPI().GetTableRequest(key).Send(ctx)
	if err != nil {
		return nil, err
	}
	
	return table.Columns, nil
```

Reply:
This way it is less nested and we know that when we do not have flag `headers` set, it returns directly.

Reply:
fixed [d827bf2](https://github.com/keboola/keboola-as-code/pull/1892/commits/d827bf2d077fd714171b52cb659415d1f123fabc)